dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs

```diff
@@ -887,6 +888,16 @@ private static ChatMessageContentPart GetAudioContentItem(AudioContent audioCont
         throw new ArgumentException($"{nameof(AudioContent)} must have Data bytes.");
     }
 
+    private static ChatMessageContentPart GetBinaryContentItem(BinaryContent binaryContent)
+    {
+        if (binaryContent.Data is { IsEmpty: false } data)
+        {
+            return ChatMessageContentPart.CreateFilePart(BinaryData.FromBytes(data), binaryContent.MimeType, Guid.NewGuid().ToString());
+        }
+
+        throw new ArgumentException($"{nameof(BinaryContent)} must have Data bytes.");
```

Comment:
Suggest adding 2 unit tests 

1. One that throws reaching this line of code 
2. One that doesn't throw, ensuring that adding a Binary Content in the ChatHistory won't throw.